diff -ruN PreLiquidation.sol PreLiquidation.sol
--- PreLiquidation.sol	2025-03-10 10:47:36.000000000 +0100
+++ PreLiquidation.sol	2025-03-10 13:38:05.000000000 +0100
@@ -20,7 +20,7 @@
 /// @author Morpho Labs
 /// @custom:contact security@morpho.org
 /// @notice A linear LIF and linear LCF pre-liquidation contract for Morpho.
-contract PreLiquidation is IPreLiquidation, IMorphoRepayCallback {
+contract PreLiquidation is IMorphoRepayCallback {
     using SharesMathLib for uint256;
     using MathLib for uint256;
     using SafeTransferLib for ERC20;
@@ -129,10 +129,14 @@
     /// once. It increases linearly from preLCF1 at preLltv to preLCF2 at LLTV.
     /// @dev The pre-liquidation incentive factor (preLIF) is the factor by which the repaid debt is multiplied to
     /// compute the seized collateral. It increases linearly from preLIF1 at preLltv to preLIF2 at LLTV.
-    function preLiquidate(address borrower, uint256 seizedAssets, uint256 repaidShares, bytes calldata data)
-        external
-        returns (uint256, uint256)
-    {
+    function preLiquidate(
+        address borrower,
+        uint256 seizedAssets,
+        uint256 repaidShares,
+        uint256 lif,
+        uint256 lcf,
+        bytes calldata data
+    ) external returns (uint256, uint256) {
         require(UtilsLib.exactlyOneZero(seizedAssets, repaidShares), ErrorsLib.InconsistentInput());
 
         MORPHO.accrueInterest(marketParams());
@@ -151,7 +155,7 @@
 
         uint256 ltv = borrowed.wDivUp(collateralQuoted);
         uint256 quotient = (ltv - PRE_LLTV).wDivDown(LLTV - PRE_LLTV);
-        uint256 preLIF = quotient.wMulDown(PRE_LIF_2 - PRE_LIF_1) + PRE_LIF_1;
+        uint256 preLIF = lif;
 
         if (seizedAssets > 0) {
             uint256 seizedAssetsQuoted = seizedAssets.mulDivUp(collateralPrice, ORACLE_PRICE_SCALE);
@@ -166,7 +170,7 @@
 
         // Note that the pre-liquidation close factor can be greater than WAD (100%).
         // In this case the position can be fully pre-liquidated.
-        uint256 preLCF = quotient.wMulDown(PRE_LCF_2 - PRE_LCF_1) + PRE_LCF_1;
+        uint256 preLCF = lcf;
 
         uint256 repayableShares = uint256(position.borrowShares).wMulDown(preLCF);
         require(repaidShares <= repayableShares, ErrorsLib.PreLiquidationTooLarge(repaidShares, repayableShares));
@@ -195,4 +199,42 @@
 
         ERC20(LOAN_TOKEN).safeTransferFrom(liquidator, address(this), repaidAssets);
     }
+
+    function preLIF(address borrower) external returns (uint256) {
+        Market memory market = MORPHO.market(ID);
+        Position memory position = MORPHO.position(ID, borrower);
+
+        uint256 collateralPrice = IOracle(PRE_LIQUIDATION_ORACLE).price();
+        uint256 collateralQuoted = uint256(position.collateral).mulDivDown(collateralPrice, ORACLE_PRICE_SCALE);
+        uint256 borrowed = uint256(position.borrowShares).toAssetsUp(market.totalBorrowAssets, market.totalBorrowShares);
+
+        // The two following require-statements ensure that collateralQuoted is different from zero.
+        require(borrowed <= collateralQuoted.wMulDown(LLTV), ErrorsLib.LiquidatablePosition());
+        // The following require-statement is equivalent to checking that ltv > PRE_LLTV.
+        require(borrowed > collateralQuoted.wMulDown(PRE_LLTV), ErrorsLib.NotPreLiquidatablePosition());
+
+        uint256 ltv = borrowed.wDivUp(collateralQuoted);
+        uint256 quotient = (ltv - PRE_LLTV).wDivDown(LLTV - PRE_LLTV);
+        uint256 preLIF = quotient.wMulDown(PRE_LIF_2 - PRE_LIF_1) + PRE_LIF_1;
+        return preLIF;
+    }
+
+    function preLCF(address borrower) external returns (uint256) {
+        Market memory market = MORPHO.market(ID);
+        Position memory position = MORPHO.position(ID, borrower);
+
+        uint256 collateralPrice = IOracle(PRE_LIQUIDATION_ORACLE).price();
+        uint256 collateralQuoted = uint256(position.collateral).mulDivDown(collateralPrice, ORACLE_PRICE_SCALE);
+        uint256 borrowed = uint256(position.borrowShares).toAssetsUp(market.totalBorrowAssets, market.totalBorrowShares);
+
+        // The two following require-statements ensure that collateralQuoted is different from zero.
+        require(borrowed <= collateralQuoted.wMulDown(LLTV), ErrorsLib.LiquidatablePosition());
+        // The following require-statement is equivalent to checking that ltv > PRE_LLTV.
+        require(borrowed > collateralQuoted.wMulDown(PRE_LLTV), ErrorsLib.NotPreLiquidatablePosition());
+
+        uint256 ltv = borrowed.wDivUp(collateralQuoted);
+        uint256 quotient = (ltv - PRE_LLTV).wDivDown(LLTV - PRE_LLTV);
+        uint256 preLCF = quotient.wMulDown(PRE_LCF_2 - PRE_LCF_1) + PRE_LCF_1;
+        return preLCF;
+    }
 }
