diff -ruN PreLiquidation.sol PreLiquidation.sol
--- PreLiquidation.sol	2024-09-19 08:12:23.000000000 +0200
+++ PreLiquidation.sol	2024-09-19 13:53:43.000000000 +0200
@@ -1,20 +1,20 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity 0.8.27;
 
-import {Id, MarketParams, IMorpho, Position, Market} from "../lib/morpho-blue/src/interfaces/IMorpho.sol";
-import {IOracle} from "../lib/morpho-blue/src/interfaces/IOracle.sol";
-import {UtilsLib} from "../lib/morpho-blue/src/libraries/UtilsLib.sol";
-import {MarketParamsLib} from "../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
-import "../lib/morpho-blue/src/libraries/ConstantsLib.sol";
-import {MathLib} from "../lib/morpho-blue/src/libraries/MathLib.sol";
-import {SharesMathLib} from "../lib/morpho-blue/src/libraries/SharesMathLib.sol";
-import {SafeTransferLib} from "../lib/solmate/src/utils/SafeTransferLib.sol";
-import {ERC20} from "../lib/solmate/src/tokens/ERC20.sol";
+import {Id, MarketParams, IMorphoHarness, Position, Market} from "../helpers/IMorphoHarness.sol";
+import {IOracle} from "../../lib/morpho-blue/src/interfaces/IOracle.sol";
+import {UtilsLib} from "../../lib/morpho-blue/src/libraries/UtilsLib.sol";
+import {MarketParamsLib} from "../../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
+import "../../lib/morpho-blue/src/libraries/ConstantsLib.sol";
+import {MathLib} from "../../lib/morpho-blue/src/libraries/MathLib.sol";
+import {SharesMathLib} from "../../lib/morpho-blue/src/libraries/SharesMathLib.sol";
+import {SafeTransferLib} from "../../lib/solmate/src/utils/SafeTransferLib.sol";
+import {ERC20} from "../../lib/solmate/src/tokens/ERC20.sol";
 import {EventsLib} from "./libraries/EventsLib.sol";
 import {ErrorsLib} from "./libraries/ErrorsLib.sol";
 import {IPreLiquidationCallback} from "./interfaces/IPreLiquidationCallback.sol";
 import {IPreLiquidation, PreLiquidationParams} from "./interfaces/IPreLiquidation.sol";
-import {IMorphoRepayCallback} from "../lib/morpho-blue/src/interfaces/IMorphoCallbacks.sol";
+import {IMorphoRepayCallback} from "../../lib/morpho-blue/src/interfaces/IMorphoCallbacks.sol";
 
 /// @title Morpho
 /// @author Morpho Labs
@@ -29,7 +29,7 @@
     using SafeTransferLib for ERC20;
 
     /* IMMUTABLE */
-    IMorpho public immutable MORPHO;
+    IMorphoHarness public immutable MORPHO;
     Id public immutable marketId;
 
     // Market parameters
@@ -45,7 +45,7 @@
     uint256 public immutable preLiquidationIncentive;
 
     constructor(MarketParams memory _marketParams, PreLiquidationParams memory _preLiquidationParams, address morpho) {
-        MORPHO = IMorpho(morpho);
+        MORPHO = IMorphoHarness(morpho);
         marketId = _marketParams.id();
 
         require(MORPHO.market(marketId).lastUpdate != 0, ErrorsLib.NonexistentMarket(marketId));
diff -ruN PreLiquidationFactory.sol PreLiquidationFactory.sol
--- PreLiquidationFactory.sol	2024-09-19 08:12:23.000000000 +0200
+++ PreLiquidationFactory.sol	2024-09-19 11:30:51.000000000 +0200
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity 0.8.27;
 
-import {IMorpho, MarketParams} from "../lib/morpho-blue/src/interfaces/IMorpho.sol";
+import {IMorphoHarness, MarketParams} from "../helpers/IMorphoHarness.sol";
 import {PreLiquidation} from "./PreLiquidation.sol";
 import {IPreLiquidation, PreLiquidationParams} from "./interfaces/IPreLiquidation.sol";
 import {ErrorsLib} from "./libraries/ErrorsLib.sol";
@@ -16,7 +16,7 @@
     /* IMMUTABLE */
 
     /// @inheritdoc IPreLiquidationFactory
-    IMorpho public immutable MORPHO;
+    IMorphoHarness public immutable MORPHO;
 
     /* STORAGE */
 
@@ -29,7 +29,7 @@
     constructor(address morpho) {
         require(morpho != address(0), ErrorsLib.ZeroAddress());
 
-        MORPHO = IMorpho(morpho);
+        MORPHO = IMorphoHarness(morpho);
     }
 
     /* EXTERNAL */
diff -ruN interfaces/IPreLiquidation.sol interfaces/IPreLiquidation.sol
--- interfaces/IPreLiquidation.sol	2024-09-19 08:12:23.000000000 +0200
+++ interfaces/IPreLiquidation.sol	2024-09-19 13:56:01.000000000 +0200
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity >= 0.5.0;
 
-import {Id, MarketParams, IMorpho} from "../../lib/morpho-blue/src/interfaces/IMorpho.sol";
+import {Id, MarketParams, IMorphoHarness} from "../../helpers/IMorphoHarness.sol";
 
 struct PreLiquidationParams {
     uint256 preLltv;
@@ -10,7 +10,7 @@
 }
 
 interface IPreLiquidation {
-    function MORPHO() external view returns (IMorpho);
+    function MORPHO() external view returns (IMorphoHarness);
 
     function marketId() external view returns (Id);
 
diff -ruN interfaces/IPreLiquidationFactory.sol interfaces/IPreLiquidationFactory.sol
--- interfaces/IPreLiquidationFactory.sol	2024-09-19 08:12:23.000000000 +0200
+++ interfaces/IPreLiquidationFactory.sol	2024-09-19 13:55:49.000000000 +0200
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity >= 0.5.0;
 
-import {MarketParams, IMorpho} from "../../lib/morpho-blue/src/interfaces/IMorpho.sol";
+import {MarketParams, IMorpho} from "../../helpers/IMorphoHarness.sol";
 import {IPreLiquidation, PreLiquidationParams} from "./IPreLiquidation.sol";
 
 /// @title IPreLiquidationFactory
@@ -10,7 +10,7 @@
 /// @notice Interface of PreLiquidation's factory.
 interface IPreLiquidationFactory {
     /// @notice The address of the Morpho contract.
-    function MORPHO() external view returns (IMorpho);
+    function MORPHO() external view returns (IMorphoHarness);
 
     /// @notice The contract address created for a specific preLiquidationId.
     function preLiquidations(bytes32) external view returns (IPreLiquidation);
diff -ruN libraries/ErrorsLib.sol libraries/ErrorsLib.sol
--- libraries/ErrorsLib.sol	2024-09-19 08:12:23.000000000 +0200
+++ libraries/ErrorsLib.sol	2024-09-19 13:51:28.000000000 +0200
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity 0.8.27;
 
-import {Id} from "../../lib/morpho-blue/src/interfaces/IMorpho.sol";
+import {Id} from "../../../lib/morpho-blue/src/interfaces/IMorpho.sol";
 
 /// @title ErrorsLib
 /// @author Morpho Labs
diff -ruN libraries/EventsLib.sol libraries/EventsLib.sol
--- libraries/EventsLib.sol	2024-09-19 08:12:23.000000000 +0200
+++ libraries/EventsLib.sol	2024-09-19 13:51:50.000000000 +0200
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity 0.8.27;
 
-import {Id, MarketParams} from "../../lib/morpho-blue/src/interfaces/IMorpho.sol";
+import {Id, MarketParams} from "../../../lib/morpho-blue/src/interfaces/IMorpho.sol";
 import {PreLiquidationParams} from "../interfaces/IPreLiquidation.sol";
 
 /// @title EventsLib
